code/_onclick/hud/alert.dm

```diff
@@ -312,21 +314,42 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
  * Handles assigning most of the variables for the alert that pops up when an item is offered
  *
  * Handles setting the name, description and icon of the alert and tracking the person giving
- * and the item being offered, also registers a signal that removes the alert from anyone who moves away from the offerer
+ * and the item being offered.
  * Arguments:
  * * taker - The person receiving the alert
  * * offerer - The person giving the alert and item
  * * receiving - The item being given by the offerer
  */
 /atom/movable/screen/alert/give/proc/setup(mob/living/carbon/taker, mob/living/carbon/offerer, obj/item/receiving)
-	name = "[offerer] is offering [receiving]"
-	desc = "[offerer] is offering [receiving]. Click this alert to take it."
+	var/receiving_name = get_receiving_name(taker, offerer, receiving)
```

Comment:
You can probably use ``replacetext`` instead, and I think it'll look way better if it did

Reply:
Is that actually better though? Like, I don't really know the metrics of replacetext, but my gut feeling tells me it's probably worse for performance, and to me this didn't feel quite that bad in terms of usage of string conversion. We do that kind of stuff nearly everywhere.

Reply:
I don't think this is something to worry about the performance of, I think replacetext would be fine to use here if just to help make it easier for people to understand.